libs/langchain/langchain/chains/base.py

```diff
@@ -232,6 +231,7 @@ def __call__(
             dumpd(self),
             inputs,
         )
+        inputs = self.prep_inputs(inputs, callbacks=run_manager.get_child())
```

Comment:
Note: this moves the `prep_inputs` to AFTER the `on_chain_start` call. I think that makes more sense anyway since then we handle the actual initial chain inputs before prepping and then the LLM or whatever child call shows the loaded memory, etc. there?